File: src/javatest/trickyquestion/Question1.java
package javatest.trickyquestion;

public class Question1 extends Object {
    public static void main(String[] args) {
     new A().print(null);
    }
}
class A{
    A() {
        System.out.println("A");
    }
    public void print(String str){
        System.out.println("String");
    }
    public void print(Object obj) {
        System.out.println("Object");
    }
}

File: src/kotlintest/trickyquestion/Question1.kt
package kotlintest.trickyquestion

class Question1 {

}

fun main(args : Array<String>){
   println((1..100).map {
       index-> mapOf(0 to index, index%3 to "Fizz",index%5 to "Buzz", index%15 to "Fizz Buzz")[0]
   })
}